src/compiler/checker.ts

```diff
@@ -26003,7 +26003,7 @@ namespace ts {
             }
 
             // Referencing abstract properties within their own constructors is not allowed
-            if ((flags & ModifierFlags.Abstract) && isThisProperty(node) && symbolHasNonMethodDeclaration(prop)) {
+            if ((flags & ModifierFlags.Abstract) && (isThisProperty(node) || isThisInitializedDeclaration(node) || isThisInitializedExpression(node)) && symbolHasNonMethodDeclaration(prop)) {

```

Comment:
can you break this into 3 lines? (or some other way of reducing the line length)

Reply:
I think `isThisInitializedDeclaration` should check that there's actually destructuring occurring. Otherwise you might end up with an error on `let x = this` (although I'm not sure that `checkPropertyAccessibility` would be called in that case)

Reply:
Done